connectivity/nanostack/sal-stack-nanostack/source/Service_Libs/fhss/fhss_ws.c

```diff
@@ -499,16 +499,35 @@ static uint32_t fhss_ws_calculate_ufsi(fhss_structure_t *fhss_structure, uint32_
         }
     }
     cur_slot--;
-    uint32_t remaining_time_ms = 0;
-    if (fhss_structure->ws->unicast_timer_running == true) {
-        remaining_time_ms = US_TO_MS(get_remaining_slots_us(fhss_structure, fhss_unicast_handler, MS_TO_US(dwell_time) - NS_TO_US((int64_t)(fhss_structure->ws->drift_per_millisecond_ns * dwell_time))));
-    }
+
     uint32_t time_to_tx = 0;
     uint32_t cur_time = fhss_structure->callbacks.read_timestamp(fhss_structure->fhss_api);
     if (cur_time < tx_time) {
         time_to_tx = US_TO_MS(tx_time - cur_time);
     }
-    uint64_t ms_since_seq_start = (cur_slot * dwell_time) + (dwell_time - remaining_time_ms) + time_to_tx;
+    uint64_t ms_since_seq_start;
+    if (fhss_structure->ws->unicast_timer_running == true) {
+        if (fhss_structure->ws->next_uc_timeout < cur_time) {
```

Comment:
If next_uc_timeout overflows, this may be true even if the timer has not yet expired

Reply:
Indeed I didn't handle rollover. Actually I just followed the logic of the time comparison that is made a few line above which seems to have the same problem (probably the whole stack?). 
if (cur_time < tx_time) {

Reply:
```suggestion
        // Allow timer interrupt to delay max 10 seconds, otherwise assume next_uc_timeout overflowed
        if ((fhss_structure->ws->next_uc_timeout < cur_time) && ((cur_time - fhss_structure->ws->next_uc_timeout) < 10000000)) {
```